core/runtime/src/main/java/io/quarkus/runtime/metrics/MetricsFactory.java

```diff
@@ -0,0 +1,141 @@
+    MetricBuilder builder(String name);
+
+    interface MetricBuilder {
+        /**
+         * @param description Description text of the eventual metric (optional).
+         * @return The builder with added description.
+         */
+        MetricBuilder description(String description);
+
+        /**
+         * @param key The tag key.
+         * @param value The tag value.
+         * @return The builder with added tag.
+         */
+        MetricBuilder tag(String key, String value);
+
+        /**
+         * Specify the metric unit (optional)
+         * 
+         * @param unit Base unit of the eventual metric
+         * @return The builder with added base unit.
+         */
+        MetricBuilder unit(String unit);
+
+        /**
+         * Register a counter that retrieves its value from a supplier function
+         * 
+         * @param countFunction Function supplying a monotonically increasing number value
+         */
+        void counter(Supplier<Number> countFunction);
+
+        /**
+         * Register a counter that retrieves its value by the applying a function
+         * to an object
+         * 
+         * @param obj Object instance to observe
+         * @param countFunction Function returning a monotonically increasing double value
+         */
+        <S> void counter(S obj, ToDoubleFunction<S> countFunction);
+
+        /**
+         * Register a gauge that retrieves its value from a supplier function
+         * 
+         * @param gaugeFunction Function supplying number value
+         */
+        void gauge(Supplier<Number> gaugeFunction);
+
+        /**
+         * Register a gauge that retrieves its value by applying a function
+         * to an object
+         * 
+         * @param obj Object instance to observe
+         * @param gaugeFunction Function returning a long value
+         */
+        <S> void gauge(S obj, ToDoubleFunction<S> gaugeFunction);
+
+        /**
+         * @return TimeRecorder to measure passage of time using
+         *         incremental updates.
+         */
+        TimeRecorder timeRecorder();
+
+        /**
+         * Wrap a {@link Runnable} so that it is timed when invoked.
+         *
+         * @param f The Runnable to time when it is invoked.
+         * @return The wrapped Runnable.
+         */
+        Runnable timed(Runnable f);
+
+        /**
+         * Wrap a {@link Callable} so that it is timed when invoked.
+         *
+         * @param f The Callable to time when it is invoked.
+         * @param <T> The return type of the callable.
+         * @return The wrapped callable.
+         */
+        <T> Callable<T> timed(Callable<T> f);
+
+        /**
+         * Wrap a {@link Supplier} so that it is timed when invoked.
+         *
+         * @param f The {@code Supplier} to time when it is invoked.
+         * @param <T> The return type of the {@code Supplier} result.
+         * @return The wrapped supplier.
+         * @since 1.2.0
```

Comment:
:no_entry:

Reply:
Better avoid emoji comments for reviews as the intent might not be clear :). And if you get a question about it, it's not going to be a win.

Reply:
Right. Also I'm not sure what you mean @jmartisk ? suggesting that the _since_ attribute should be removed?

Reply:
Ok, noted. I'm saying that this is clearly not since Quarkus 1.2.0, so either the version number should be corrected, or the attribute removed :)

Reply:
HA! Yes. I can remove that. Thanks for asking for emoji clarity @gsmet and @Sanne =)